packages/jest-types/src/Global.ts

```diff
@@ -55,20 +55,63 @@ export type EachTestFn<EachCallback extends TestCallback> = (
   ...args: ReadonlyArray<any>
 ) => ReturnType<EachCallback>;
 
-type Each<EachCallback extends TestCallback, Name> =
-  | ((
-      table: EachTable,
-      ...taggedTemplateData: TemplateData
-    ) => (name: Name, test: EachTestFn<EachCallback>, timeout?: number) => void)
-  | (() => () => void);
+interface Each<EachFn extends TestFn | BlockFn> {
+  <T extends Record<string, unknown>>(table: ReadonlyArray<T>): (
+    name: string | NameLike,
+    fn: (arg: T) => ReturnType<EachFn>,
+    timeout?: number,
+  ) => void;
+
+  <T extends readonly [unknown, ...Array<unknown>]>(table: ReadonlyArray<T>): (
```

Comment:
An array of arrays is actually treated as array of tuples. Spreading `['abc', 123]` as tuple gives a type `a: string, b: number`, spreading the same thing as an array results in a type `a: string | number, b: string | number`. The hard part was to capture indefinite number of tuple members. `[unknown, ...Array<unknown>]` looks clumsy, but does this job well.